cocos2d/core/renderer/assembler-2d.js

```diff
@@ -125,7 +125,7 @@ export default class Assembler2D extends Assembler {
         let ibuf = buffer._iData,
             indiceOffset = offsetInfo.indiceOffset,
             vertexId = offsetInfo.vertexOffset;
-        for (let i = 0, l = iData.length; i < l; i++) {
+        for (let i = 0, l = this.indicesCount; i < l; i++) {
             ibuf[indiceOffset++] = vertexId + iData[i];
```

Comment:
It may exceed iData too. So i think should use the min value of them, for example
```typescript
const indicesToFill = min(this.indicesCount, iData.length);
```